server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -374,12 +377,15 @@ private String getChecksumOfLocalFile(String file) throws IOException {
         return localSegmentChecksumMap.get(file);
     }
 
-    private void deleteStaleCommits() {
-        try {
-            remoteDirectory.deleteStaleSegments(LAST_N_METADATA_FILES_TO_KEEP);
-        } catch (IOException e) {
-            logger.info("Exception while deleting stale commits from remote segment store, will retry delete post next commit", e);
-        }
+    private void deleteStaleCommitsAsync(String threadpoolName, Runnable onCompletion) {
+        indexShard.getThreadPool().executor(threadpoolName).execute(() -> {
```

Comment:
`indexShard.getThreadPool().executor(threadpoolName).execute(Runnable)` can throw exceptions too. Lets handle this with a catch block and run onCompletion with the catch block.